src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/AbstractBuildEventValueProvider.Helper.cs

```diff
@@ -0,0 +1,136 @@
+            {
+                if (string.IsNullOrWhiteSpace(unevaluatedPropertyValue) &&
+                    !unevaluatedPropertyValue.Contains("\n"))
+                {
+                    var result = FindTargetToRemove(projectXml);
+                    if (result.success)
+                    {
+                        projectXml.RemoveChild(result.target);
+                    }
+                }
+                else
+                {
+                    SetParameter(projectXml, unevaluatedPropertyValue);
+                }
+            }
+
+            private (bool success, ProjectTaskElement execTask) FindExecTaskInTargets(ProjectRootElement projectXml)
+            {
+                var execTask = projectXml.Targets
+                                    .Where(target => GetTargetString(target) == BuildEventString)
+                                    .SelectMany(target => target.Tasks)
+                                    .Where(task => task.Name == _execTaskName)
+                                    .FirstOrDefault();
+                return (success: execTask != null, execTask: execTask);
+            }
+
+            private (bool success, ProjectTargetElement target) FindTargetToRemove(ProjectRootElement projectXml)
+            {
+                var targetArray = projectXml.Targets
+                                        .Where(target =>
+                                            GetTargetString(target) == BuildEventString &&
+                                            target.Children.Count == 1 &&
+                                            target.Tasks.Count == 1 &&
+                                            target.Tasks.First().Name == _execTaskName);
+                return (success: targetArray.Count() == 1, target: targetArray.SingleOrDefault());
+            }
+
+            private void SetParameter(ProjectRootElement projectXml, string unevaluatedPropertyValue)
+            {
+                var result = FindExecTaskInTargets(projectXml);
+
+                if (result.success == true)
+                {
+                    SetExecParameter(result.execTask, unevaluatedPropertyValue);
+                }
+                else
+                {
+                    var targetName = GetTargetName(projectXml);
+                    var target = projectXml.AddTarget(targetName);
+                    SetTargetDependencies(target);
+                    var execTask = target.AddTask(_execTaskName);
+                    SetExecParameter(execTask, unevaluatedPropertyValue);
+                }
+            }
+
+            private void SetExecParameter(ProjectTaskElement execTask, string unevaluatedPropertyValue)
+                => execTask.SetParameter(_commandString, unevaluatedPropertyValue);
+
+            private string GetTargetName(ProjectRootElement projectXml)
+            {
+                var targetNames = projectXml.Targets.Select(t => t.Name).ToArray();
```

Comment:
We are going to allocate a string for all targets. `.Contains` on the array wont be efficient as well. Instead if we use Hashset then the search is a constant operation